src/bot/slack_bot.py

```diff
@@ -45,38 +49,91 @@ def handle_reaction_added(event, say, client):
     if reaction not in json.loads(VALID_REACTIONS):
         return
 
+    user = event.get("user")
+    if user not in reactions_dict.keys():
+        reactions_dict[user] = (
+            event.get("event_ts"),
+            reaction,
+        )
+
     if not timer_active:
-        start_timer(client, say, event)
+        start_timer(client, say, event, isAnarchyMode())
 
 
-def start_timer(client, say, event):
+def start_timer(client, say, event, anarchy_mode_active):
+    global reactions_dict
     global timer_active
+    global inputs_to_save
 
     timer_active = True
+
     print(f"{TIMER_DURATION} second timer started...")
     time.sleep(TIMER_DURATION)
     print(f"{TIMER_DURATION} second timer complete!")
 
+    if anarchy_mode_active:
+        try:
+            print("Anarchy mode is active")
+            handle_input(
+                event,
+                say,
+                client,
+                post_delete_actions_callback,
+                True,
+                anarchy_inputs=convert_reactions_dict_to_input_list(reactions_dict),
+            )
+        except errors.SlackApiError as e:
+            print(e)
+
+    else:
+        button = calculate_reactions(client, say, event)
+        try:
+            handle_input(
+                event, say, client, post_delete_actions_callback, False, button
+            )
+        except errors.SlackApiError as e:
+            print(e)
+
+    post_cycle_actions(inputs_to_save, anarchy_mode_active)
+
+
+def post_delete_actions_callback(reactions_to_save: list[str]):
+    global timer_active
+    global reactions_dict
+    global inputs_to_save
+
     timer_active = False
+    inputs_to_save = reactions_to_save
+    reactions_dict = {}
+
+
+def convert_reactions_dict_to_input_list(reactions: dict):
+    sorted_values = sorted(reactions.values(), key=lambda x: x[0])
 
-    button = calculate_reactions(client, say, event)
-    try:
-        handle_input(event, say, client, button)
-    except errors.SlackApiError as e:
-        print(e)
-    post_cycle_actions(button)
+    return [t[1] for t in sorted_values]
 
 
-def post_cycle_actions(button):
-    write_inputs_to_file(button)
+def post_cycle_actions(inputs_to_save: list[str], anarchy_mode_active: bool):
+    write_inputs_to_file(inputs_to_save, anarchy_mode_active)
 
 
-def write_inputs_to_file(button):
+def write_inputs_to_file(inputs_to_save: list[str], anarchy_mode_active: bool):
     if os.path.exists("data/inputs.csv"):
         with open("data/inputs.csv", "r", newline="") as file:
             reader = csv.reader(file)
             row_count = sum(1 for row in reader) - 1
 
     with open("data/inputs.csv", "a", newline="") as file:
         writer = csv.writer(file)
-        writer.writerow([row_count, button])
+        for index, button in enumerate(inputs_to_save):
+            writer.writerow(
+                [row_count + index, button, 200 if anarchy_mode_active else 500]
+            )
+
+
+def isAnarchyMode():
```

Comment:
Let's not use this. I'd prefer we ultimately have a [Slash Command](https://api.slack.com/interactivity/slash-commands) to toggle anarchy mode. I don't like this being hardcoded like this.

Reply:
Makes sense. I can look to add the slash command into this? Will need to check how to do it such that nobody authorised is allowed to alter the mode

My other thought was to set an env var on it or smth

Reply:
Yeah I'd rather it require the user to be authorised, even if that means it's a series of slack IDs in env vars